pmd-core/src/test/java/net/sourceforge/pmd/processor/MultiThreadProcessorTest.java

```diff
@@ -4,79 +4,78 @@
 import net.sourceforge.pmd.lang.rule.AbstractRule;
-import net.sourceforge.pmd.renderers.AbstractAccumulatingRenderer;
-import net.sourceforge.pmd.renderers.Renderer;
+import net.sourceforge.pmd.reporting.FileAnalysisListener;
+import net.sourceforge.pmd.reporting.GlobalAnalysisListener;
 import net.sourceforge.pmd.util.datasource.DataSource;
-import net.sourceforge.pmd.util.datasource.internal.AbstractDataSource;
 
 public class MultiThreadProcessorTest {
 
-    private RuleContext ctx;
-    private MultiThreadProcessor processor;
-    private RuleSets ruleSets;
+    private GlobalAnalysisListener listener;
+
     private List<DataSource> files;
     private SimpleReportListener reportListener;
+    private PMDConfiguration configuration;
 
-    public void setUpForTest(final String ruleset) {
-        PMDConfiguration configuration = new PMDConfiguration();
+    public RuleSets setUpForTest(final String ruleset) {
+        configuration = new PMDConfiguration();
         configuration.setThreads(2);
-        files = new ArrayList<>();
-        files.add(new StringDataSource("file1-violation.dummy", "ABC"));
-        files.add(new StringDataSource("file2-foo.dummy", "DEF"));
+        files = listOf(
+            DataSource.forString("abc", "file1-violation.dummy"),
+            DataSource.forString("DEF", "file2-foo.dummy")
+        );
 
         reportListener = new SimpleReportListener();
-        ctx = new RuleContext();
-        ctx.getReport().addListener(reportListener);
+        listener = GlobalAnalysisListener.tee(listOf(
+            new GlobalReportBuilderListener(),
+            reportListener
+        ));
 
-        processor = new MultiThreadProcessor(configuration);
-        ruleSets = new RuleSets(new RuleSetLoader().loadFromResources(ruleset));
+        return new RuleSets(new RuleSetLoader().loadFromResource(ruleset));
     }
 
-    @Test
-    public void testRulesDysnfunctionalLog() throws IOException {
-        setUpForTest("rulesets/MultiThreadProcessorTest/dysfunctional.xml");
-        final SimpleRenderer renderer = new SimpleRenderer(null, null);
-        renderer.start();
-        processor.processFiles(ruleSets, files, ctx, Collections.<Renderer>singletonList(renderer));
-        renderer.end();
-
-        final Iterator<ConfigurationError> configErrors = renderer.getReport().getConfigurationErrors().iterator();
-        final ConfigurationError error = configErrors.next();
-
-        Assert.assertEquals("Dysfunctional rule message not present",
-                DysfunctionalRule.DYSFUNCTIONAL_RULE_REASON, error.issue());
-        Assert.assertEquals("Dysfunctional rule is wrong",
-                DysfunctionalRule.class, error.rule().getClass());
-        Assert.assertFalse("More configuration errors found than expected", configErrors.hasNext());
-    }
+    // Dysfunctional rules are pruned upstream of the processor.
+    //
+    //    @Test
+    //    public void testRulesDysnfunctionalLog() throws Exception {
```

Comment:
I assume, we don't have a test yet to verify, that we remove broken rules... then let's keep this test in as commented as a reminder